components/input/gpio/gpio_test.go

```diff
@@ -354,9 +449,9 @@ func TestGPIOInput(t *testing.T) {
 		iterations := 50
 
 		for i := 0; i < iterations; i++ {
-			err := fakeboard.Tick(s.ctx, s.interrupt2, true, uint64(time.Now().UnixNano()))
+			err := s.interrupt2.Tick(context.Background(), true, uint64(time.Now().UnixNano()))
 			test.That(t, err, test.ShouldBeNil)
-			err = fakeboard.Tick(s.ctx, s.interrupt2, false, uint64(time.Now().UnixNano()))
+			err = s.interrupt2.Tick(context.Background(), false, uint64(time.Now().UnixNano()))
```

Comment:
[q] I don't see a strong reason to use s.ctx other than it' there, but why change it? has it been cancelled by this point for some reason?

Reply:
Yeah i shoudve used s.ctx thanks for pointing this out! changed